generators/csharp/codegen/src/ast/Method.ts

```diff
@@ -90,23 +90,16 @@ export class Method extends AstNode {
 
     public write(writer: Writer): void {
         if (this.summary != null) {
-            writer.writeLine("/// <summary>");
-            this.summary.split("\n").forEach((line) => {
-                writer.writeLine(`/// ${line}`);
-            });
-
-            writer.writeLine("/// </summary>");
+            writer.writeDocXml().writeNodeWithEscaping("summary", this.summary);
         }
         if (this.codeExample != null) {
-            writer.writeLine("/// <example>");
-            writer.writeLine("/// <code>");
-            this.codeExample.split("\n").forEach((line) => {
-                if (line !== "") {
-                    writer.writeLine(`/// ${line}`);
-                }
-            });
-            writer.writeLine("/// </code>");
-            writer.writeLine("/// </example>");
+            writer
```

Comment:
The `DocXmlWriter ` is awesome! 

and overall i like the ideas that you're bringing here, such as this fluent writer API, but i'd prefer consistency with our existing writer. I'd also like to avoid adding new `write` methods to our existing writer. So, I have a couple changes to propose:

1. remove the `writeDocXml` method from the `Writer` class and instead just initialize a `DocXmlWriter` here (and elsewhere)
2. remove the fluent elements of the `DocXmlWriter` by simply have the methods return `void`, and changing the usages accordingly

I ack this is fairly opinionated, but I think we consider it more important to adhere to a consistent style than to introduce new patterns that bring fairly slight improvements

Reply:
Done!